### PR TITLE
registration implementation + other implementation hints

### PR DESCRIPTION
**Pull request by doodyparizada**

### CODE REVIEW

File: scripts/src/models/offers.ts [hunk 1]
```diff
@@ -9,137 +9,97 @@ export type OfferMeta = {
...
 @Register
-@Index(["_offerId", "_content"], { unique: true })
 export class OfferContent extends Model {
 	@PrimaryColumn({ name: "offer_id" })
-	private _offerId: string;
+	public offerId: string;
 
-	@PrimaryColumn({ name: "content" })
-	private _content: string;
+	@Column("simple-json")
```
Comment (nitzantomer) [inline scripts/src/models/offers.ts]: are we sure that it is a json?

File: scripts/src/models/users.ts [hunk 2]
```diff
@@ -15,48 +15,60 @@ export class User extends CreationDateModel {
...
+		// XXX token could be a JWT
+		Object.assign(this, { expireDate, userId, deviceId, valid });
+	}
+
+	public isExpired(): boolean {
+		return this.expireDate > new Date();
 	}
+
+	public isAboutToExpire(): boolean {
+		const dayFromNow = new Date();
```
Comment (nitzantomer) [inline scripts/src/models/users.ts]: https://momentjs.com/

File: scripts/src/models/offers.ts [hunk 3]
```diff
@@ -9,137 +9,97 @@ export type OfferMeta = {
...
-
-	public get ownerId(): string {
-		return this._ownerId;
-	}
 }
 
-@Entity()
+@Entity({ name: "offer_contents" })
 @Register
-@Index(["_offerId", "_content"], { unique: true })
```
[the remark this answers is not in the document]
Comment (doodyparizada) [reply scripts/src/models/offers.ts]: don't need an index on the content

File: scripts/src/models/offers.ts (hunk 1 above)
Comment (doodyparizada) [reply scripts/src/models/offers.ts]: maybe not. For now it is

File: scripts/src/create.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	await (new AuthToken(user1.id, "device1", true)).save();
+	await (new AuthToken(user2.id, "device2", true)).save();
+
+	const app: Application = await (new Application("kik", "jwt")).save();
+
+	const owner = new OfferOwner();
+	owner.name = "donuts";
+	await owner.save();
+
+	const offer = new Offer();
```
[the remark this answers is not in the document]
Comment (doodyparizada) [reply scripts/src/create.ts]: sure - different PR?

**Pull request merged**